README.md

```diff
@@ -1,3 +1,594 @@
+
+If you believe the optimizer may be changing the behavior of your code please test with the optimizer disabled to verify and discuss with the team.
+
+### Reporting Vulnerabilities
+If you believe you've identified a security vulnerability in the Eco Currency contracts or other software, please submit to the Immunefi bounty (link coming soon) or join the Eco Association Discord (https://discord.eco.org) and tag or message an Eco Association team member.
+
+## Background
+The currency deployed here is the Optimism version of ECO, the mainnet implementation and associated governance can be found [here](https://github.com/helix-foundation/currency). The bridges can be used to move value between the Ethereum mainnet system and the Optimism one. 
+
+OP-ECO as it stands today only retains the ECO contract from mainnet, as its primary purpose is to give users the ability to transfer ECO cheaply and efficiently. OP-ECO does not have its own community governance - an abbreviated set of upgrades can be made by using the Mainnet governance system and then propagating those changes via the bridge. OP-ECO's monetary policy has been pared down to only one lever, linear inflation, and that rate is pegged to the rate set on the mainnet implementation by elected trustees. Inflation rate changes made on mainnet propagate to OP-ECO via the bridge.
+
+## Install
+To use the code you'll need the proper tools. Make sure you have a recent version of [Node.JS](https://nodejs.org), and a recent version of [NPM](https://npmjs.com).
+
+Once Node and NPM are installed you can use the `npm` command to install additional dependencies:
+```
+npm ci
+```
+
+## Usage
+These contracts are intended for deployment to the Ethereum and Optimism chains (L1EcoBridge to Eth mainnet and the rest to OP). Once deployed, you can interact with the contracts using the standard Ethereum RPC mechanisms.
+
+### Running the Linter, Tests and Coverage Report
+The commands below provide the basics to get set up developing as well as outlining conventions and standards - all code should pass the linter and prettier, all tests should pass, and code coverage should not decrease.
+
+#### Linting + prettier
+`eslint` and `solhint` are used to lint the code in this repository. Additionally, the prettier enforces a clean code style for readability. You can run the linter and prettier using:
+```
+npm run lint
+```
+and
+```
+npm run format
+``` 
+respectively. 
+
+#### Testing
+You can run the test suite by invoking:
+```
+npm run test
+```
+
+The test suite is extensive and can take some time to run.
+
+#### Coverage Reporting
+Coverage reports are generated separated for Solidity and JavaScript code:
+```
+npm run coverage:js
+npm run coverage:sol
+```
+
+Or, aliased for convenience when running both:
+```
+npm run coverage
+```
+
+### Running a deployment
```

Comment:
For the deploy it's worth mentioning that you need to copy .env.test to .env and set your INFURA_ID for  the networks deployed to.
Also may be useful to mention a faucet for funding (optional)